packages/nouns-contracts/contracts/governance/NounsDAODelegation.sol

```diff
@@ -22,19 +22,31 @@ import { NounDelegationToken } from './NounDelegationToken.sol';
 
 library NounsDAODelegation {
     function isDelegate(address account, uint256[] memory tokenIds) external view returns (bool) {
+        NounDelegationToken dt = NounDelegationToken(ds().delegationToken);
+        require(
+            dt.lastTransferTimestamp(account) < block.timestamp,
+            'cannot use voting power updated in the current block'
+        );
+
         for (uint256 i = 0; i < tokenIds.length; i++) {
-            if (delegateOf(tokenIds[i]) != account) {
+            if (delegateOf(tokenIds[i], dt) != account) {
                 return false;
             }
         }
         return true;
     }
 
-    function delegateOf(uint256 tokenId) public view returns (address) {
-        address delegationOwner = NounDelegationToken(ds().delegationToken).ownerOfNoRevert(tokenId);
+    function delegateOf(uint256 tokenId, NounDelegationToken dt) public view returns (address) {
```

Comment:
capturing a potential attack we need to discuss:

I can prevent you from voting/proposing, by watching when you're trying to vote/propose and sending you a NounDelegationToken in the same block before you vote. Then I can transfer it back to myself.
This only costs me gas, but it will fail you attempts.

Reply:
one way to prevent this, is to keep the transfer info per token id, not per account.
this could be packed into the _owners data ("extraData" in solady erc721).
I think this won't add any read or write costs because when we transfer tokens, we need to update the owner anyway. and when we check for the token owner, we read this slot.

Reply:
solved in david's PR #6